lib/mix/tasks/docs.ex

```diff
@@ -58,6 +58,14 @@ defmodule Mix.Tasks.Docs do
         ]
       end
 
+  > #### Tip {: .info}
+  >
+  > If you set `:source_url` to a GitHub/GitLab/BitBucket repo then whenever you
+  > publish a release you should publish the release to the repo (which will
+  > push a tag). If the release exists in the repo then ex_doc will generate
+  > links to the specific version of the code that the user is currently
+  > browsing.
```

Comment:
```suggestion
  > publish a new version of your package, you should run `git tag vVERSION`
  > and push the tag. This way, ExDoc will generate links to the specific version
  > the docs were generated for.
```

However, I don't think this is automatic, you must also configure the `:source_ref` option.

Reply:
Ah, yeah you're correct. I have that in so many projects (nearly every project I've forked into my `dev/forks` directory) that I forgot it was not automatic. Also it is something that works on 95% of libraries that I see on hexdocs. I also got thrown off by the explicit github/gitlab/bitbucket support on these lines (I didn't look too closely): https://github.com/elixir-lang/ex_doc/blob/8552c492fd35498c29983ac196988723bdc09e96/lib/ex_doc/config.ex#L284

I'll update the docs to add it. But it seems like it's outgrowing an admonition and will now need a new section specifically about setting up `source_url` and `source_ref`.

Reply:
A custom section looks great, yeah.